### PR TITLE
Refs Task #378597 Add payload

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets

- [#378597](https://dev.sun-asterisk.com/issues/378597)

## What's this PR do ?

- [x] Add payload

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*
image

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

File: app/Repositories/Interfaces/PayloadRepositoryInterface.php [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+namespace App\Repositories\Interfaces;
+
```
Comment (framgiaciv3) [inline app/Repositories/Interfaces/PayloadRepositoryInterface.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Classes.ClassDeclaration.OpenBraceNotAlone</br>**Severity**: error</br>**File**: app/Repositories/Interfaces/PayloadRepositoryInterface.php</br>**Line**: 6</br>Opening interface brace must be on a line by itself

Comment (framgiaciv3) [inline app/Repositories/Interfaces/PayloadRepositoryInterface.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ScopeClosingBrace.ContentBefore</br>**Severity**: error</br>**File**: app/Repositories/Interfaces/PayloadRepositoryInterface.php</br>**Line**: 6</br>Closing brace must be on a line by itself

File: app/Http/Controllers/PayloadController.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(PayloadCreateRequest $request)
+    {
+        $data = $request->only(['webhook_id', 'content']);
+        DB::beginTransaction();
+        try {
+            $payload = $this->payloadRepository->create($data);
+            $conditions = $request->only('conditions')['conditions'];
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 58</br>String "condition" does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 58</br>String "payload_id" does not require double quotes; use single quotes instead

File: app/Http/Requests/PayloadCreateRequest.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'webhook_id' => 'required',
+            'content' => 'required',
+        ];
+    }
```
Comment (framgiaciv3) [inline app/Http/Requests/PayloadCreateRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Requests/PayloadCreateRequest.php</br>**Line**: 32</br>Opening brace should be on a new line

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr></table>

File: app/Repositories/Interfaces/PayloadRepositoryInterface.php [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+namespace App\Repositories\Interfaces;
```
Comment (framgiaciv3) [inline app/Repositories/Interfaces/PayloadRepositoryInterface.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Classes.ClassDeclaration.OpenBraceNewLine</br>**Severity**: error</br>**File**: app/Repositories/Interfaces/PayloadRepositoryInterface.php</br>**Line**: 5</br>Opening brace of a interface must be on the line after the definition

Comment (framgiaciv3) [inline app/Repositories/Interfaces/PayloadRepositoryInterface.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ScopeClosingBrace.ContentBefore</br>**Severity**: error</br>**File**: app/Repositories/Interfaces/PayloadRepositoryInterface.php</br>**Line**: 5</br>Closing brace must be on a line by itself

File: app/Http/Controllers/PayloadController.php [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 37</br>Variable "webhook_id" is not in valid camel caps format

File: app/Http/Controllers/PayloadController.php [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        // $webhooks = Webhook::pluck('id', 'name');
+        return view('payloads.create', compact('webhook_id'));
+    }
+
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 50</br>Variable "webhook_id" is not in valid camel caps format

File: app/Http/Controllers/PayloadController.php [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+     */
+    public function store(PayloadCreateRequest $request, $webhook_id)
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 52</br>Variable "webhook_id" is not in valid camel caps format

**Comment by framgiaciv3** [comment]
No data coverage!

**Comment by framgiaciv3** [comment]
No data coverage!

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/PayloadController.php [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    public function store(PayloadCreateRequest $request, $webhookId)
+    {
+        $webhook = Webhook::findOrFail($webhookId);
+        $data = $request->only(['content']);
+        $data['webhook_id'] = $webhook->id;
+
+        DB::beginTransaction();
+        try {
+            $payload = $this->payloadRepository->create($data);
+            $conditions = $request->only('conditions');
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 60</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 60</br>Expected 1 space(s) after closing parenthesis; found 0

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/PayloadController.php (hunk 8 above)
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 60</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 60</br>Expected 1 space(s) after closing parenthesis; found 0

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/PayloadController.php [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+            if ($conditions) {
+                foreach ($conditions['conditions'] as $condition) {
+                    Condition::create(['condition' => $condition, 'payload_id' => $payload->id]);
```
Comment (ghost) [inline app/Http/Controllers/PayloadController.php]: `$payload->conditions()->create(['condition' => $condition]);`

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr></table>

**Review by nguyenthinhhn — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**